homeassistant/helpers/selector.py

```diff
@@ -295,25 +320,172 @@ class StringSelector(Selector):
 
     selector_type = "text"
 
-    CONFIG_SCHEMA = vol.Schema({vol.Optional("multiline", default=False): bool})
+    STRING_TYPES = [
+        "number",
+        "text",
+        "search",
+        "tel",
+        "url",
+        "email",
+        "password",
+        "date",
+        "month",
+        "week",
+        "time",
+        "datetime-local",
+        "color",
+    ]
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional("multiline", default=False): bool,
+            vol.Optional("suffix"): str,
+            # The "type" controls the input field in the browser, the resulting
+            # data can be any string so we don't validate it.
+            vol.Optional("type"): vol.In(STRING_TYPES),
+        }
+    )
 
     def __call__(self, data: Any) -> str:
         """Validate the passed selection."""
-        text = cv.string(data)
+        text: str = vol.Schema(str)(data)
         return text
 
 
+select_option = vol.All(
+    dict,
+    vol.Schema(
+        {
+            vol.Required("value"): str,
+            vol.Required("label"): str,
+        }
+    ),
+)
+
+
 @SELECTORS.register("select")
 class SelectSelector(Selector):
     """Selector for an single-choice input select."""
 
     selector_type = "select"
 
     CONFIG_SCHEMA = vol.Schema(
-        {vol.Required("options"): vol.All([str], vol.Length(min=1))}
+        {
+            vol.Required("options"): vol.All(
+                vol.Any([str], [select_option]), vol.Length(min=1)
+            )
+        }
     )
 
     def __call__(self, data: Any) -> Any:
         """Validate the passed selection."""
-        selected_option = vol.In(self.config["options"])(cv.string(data))
-        return selected_option
+        if isinstance(self.config["options"][0], str):
+            options = self.config["options"]
+        else:
+            options = [option["value"] for option in self.config["options"]]
+        return vol.In(options)(vol.Schema(str)(data))
+
+
+@SELECTORS.register("attribute")
+class AttributeSelector(Selector):
+    """Selector for an entity attribute."""
+
+    selector_type = "attribute"
+
+    CONFIG_SCHEMA = vol.Schema({vol.Required("entity_id"): cv.entity_id})
+
+    def __call__(self, data: Any) -> str:
+        """Validate the passed selection."""
+        attribute: str = vol.Schema(str)(data)
+        return attribute
+
+
+@SELECTORS.register("duration")
+class DurationSelector(Selector):
+    """Selector for a duration."""
+
+    selector_type = "duration"
+
+    CONFIG_SCHEMA = vol.Schema({})
+
+    def __call__(self, data: Any) -> timedelta:
+        """Validate the passed selection."""
+        duration: timedelta = cv.time_period_dict(data)
+        return duration
+
+
+@SELECTORS.register("icon")
+class IconSelector(Selector):
+    """Selector for an icon."""
+
+    selector_type = "icon"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {vol.Optional("placeholder"): str, vol.Optional("fallbackPath"): str}
```

Comment:
Should `fallbackPath` be camel case?

Reply:
This is how it's defined in frontend.
@bramkragten can we make it snake case instead?

Reply:
This should not be supported in the backend, should be frontend only

Reply:
Thanks, removed in 4bb9711